jgrapht-core/src/test/java/org/jgrapht/alg/cycle/ChinesePostmanTest.java

```diff
@@ -0,0 +1,325 @@
+/*
+ * (C) Copyright 2017-2017, by Joris Kinable and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.cycle;
+
+import org.jgrapht.Graph;
+import org.jgrapht.GraphPath;
+import org.jgrapht.Graphs;
+import org.jgrapht.alg.interfaces.EulerianCycleAlgorithm;
+import org.jgrapht.graph.*;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * @author Joris Kinable
+ */
+public class ChinesePostmanTest {
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testGraphNoVertices(){
+        Graph<Integer, DefaultEdge> g = new DefaultDirectedGraph<>(DefaultEdge.class);
+        EulerianCycleAlgorithm<Integer, DefaultEdge> alg=new ChinesePostman<>();
+        alg.getEulerianCycle(g);
+    }
+
+    @Test
+    public void testGraphNoEdges(){
+        Graph<Integer, DefaultEdge> g = new DefaultDirectedGraph<>(DefaultEdge.class);
```

Comment:
Doesn't this testcase fall under undefined behavior since the input graph is supposed to be (strongly) connected?  If we're not going to run a connectivity inspector and throw an IllegalArgumentException for this, then there's not much point having a test for its (incorrect) answer.